frontend/src/pages/user/admin/manageUsers/index.tsx

```diff
@@ -1,10 +1,72 @@
 import { Col, Row } from 'react-bootstrap';
-import UserTableRow from '@components/UserTableRow';
-import datas from '@pages/user/admin/UserData.json';
+import { useMutation } from '@tanstack/react-query';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { CheckMutateStatus, RouteOnNotOK } from '@lib/Status';
 import Pagination from '@components/Pagination';
+import UserTableRow from '@components/UserTableRow';
 import UserTableHeader from '@components/UserTableHeader';
+import { useEffect, useState } from 'react';
+
+interface ICreditCard {
+  CVV: string;
+  name: string;
+  card_number: string;
+  expiry_date: string;
+}
+
+interface IUser {
+  address: string;
+  credit_card: [ICreditCard];
+  email: string;
+  enabled: true;
+  icon_url: string;
+  name: string;
+  role: string;
+  username: string;
+}
 
 const ManageUser = () => {
+  const navigate = useNavigate();
+  const [isMore, setIsMore] = useState(true);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const itemLimit = 10;
+
+  const {
+    data: fetchedData,
+    status,
+    mutate,
+  } = useMutation({
+    mutationKey: ['adminGetUser'],
+    mutationFn: async (isPage: boolean) => {
+      if (!isPage) {
+        const newSearchParams = new URLSearchParams({ offset: '0', limit: itemLimit.toString() });
+        setSearchParams(newSearchParams, { replace: true });
+      }
+      const resp = await fetch('/api/admin/user?' + searchParams.toString(), {
+        method: 'GET',
+        headers: { accept: 'application/json' },
+      });
+      RouteOnNotOK(resp, navigate);
+      const response = await resp.json();
+      setIsMore(response.length === itemLimit);
+      return response as IUser[];
+    },
+  });
+
```

Comment:
When the `response.length === itemLimit`, it doesn't need add one more page

Reply:
No.
Lets says ``itemLimit`` is 10. So everytime i request for 10 items. 
- if i got has 9 or less items
  - means there's no more. Block the page component for request the next batch.
- if i got 10 items
  - means it can be the last 10 item or there're actually many more after these.
  - frontend doesnt know which is the case. so dont block page component.

Reply:
How about query with limit 11 and you will know next page is needed or not.

Reply:
Yeah i can do that.
And i need to discard the 11th one if it got one